### PR TITLE
Integrate a unified CI gate/check job

### PR DESCRIPTION
**Pull request by webknjaz**
https://github.com/marketplace/actions/alls-green#why

##### SUMMARY

This job allows reducing the maintenance burden of having to remember to manually update the branch protection on any matrix changes.

##### ISSUE TYPE

- Maintenance Pull Request
- Testing Pull Request

##### COMPONENT NAME

GHA

##### ADDITIONAL INFORMATION

N/A

### CODE REVIEW

**Comment by webknjaz** [comment]
@felixfontein ^

**Comment by felixfontein** [comment]
I'm not a fan of forcing all CI jobs to pass before merging, especially when they heavily depend on tools outside that repo. From that POV, I don't think this should be a part of the template. Whether to add such a gate/check job to a CI workflow is something every collection has to decide on their own IMO.

**Comment by briantist** [comment]
> I'm not a fan of forcing all CI jobs to pass before merging, especially when they heavily depend on tools outside that repo. From that POV, I don't think this should be a part of the template. Whether to add such a gate/check job to a CI workflow is something every collection has to decide on their own IMO.

I do agree we should not force all jobs to pass, and that it should be up to each collection. When I looked into this Action though, it does seem that it is customizable to which jobs can fail or be skipped, etc.

I suppose that also anyone using the template could remove this job completely.

One thing I don't like about including it here is that it requires repo settings to be changed before it can work as intended.
And I'll note that for maintainers like me who have their collection in `ansible-collections` but don't have full rights, I cannot change those settings without asking someone to (I can't even see them).

Maybe we should get some more community input on it?

**Comment by webknjaz** [comment]
> I'm not a fan of forcing all CI jobs to pass before merging, especially when they heavily depend on tools outside that repo.

This is solvable by allowing certain jobs to fail. It shouldn't be a big deal. Plus, it's adjustable anyway. I think it's a good default to have.
Besides, it's not enforced unless the end-users actually add it to the branch protection. So I would say this point is not a concern at all.

**Comment by webknjaz** [comment]
> One thing I don't like about including it here is that it requires repo settings to be changed before it can work as intended.

Meaning that after the initial setup the maintainers would be able to adjust what's influencing the status w/o needing somebody with admin privileges. So it can be turned into an advantage, depending on the PoV.

**Comment by webknjaz** [comment]
I didn't mention this explicitly but this sort of a status aggregator job is useful for adding more CI stages too. For example, when I implement publishing packages, I usually make those deployment jobs depend on this check. So it's value isn't only connected to the repo settings.

File: .github/workflows/ansible-test.yml
Comment (felixfontein) [inline .github/workflows/ansible-test.yml:9]:
```suggestion
# 4. If you want to prevent merging PRs that do not pass all tests,
#    make sure to add the "check" job to your repository branch
#    protection once this workflow is added.
```

File: .github/workflows/ansible-test.yml
Comment (felixfontein) [inline .github/workflows/ansible-test.yml:172]:
```suggestion
  check:  # This job does nothing and is only used for the branch protection
          # or multi-stage CI jobs, like making sure that all tests pass before
          # a publishing job is started.
```

File: .github/workflows/ansible-test.yml
Comment (webknjaz) [reply .github/workflows/ansible-test.yml]: Thanks! I think I'll improve this further by saying that a more fine-grained job voting status control is possible too.

File: .github/workflows/ansible-test.yml
Comment (webknjaz) [inline .github/workflows/ansible-test.yml:10]:
```suggestion
#    protection once this workflow is added.
#    It is also possible to tweak which jobs are allowed to fail. See
#    https://github.com/marketplace/actions/alls-green#gotchas for more detail.
```

**Comment by webknjaz** [comment]
@briantist @felixfontein the suggestions have been accepted + I've made a small addition about fine-grained tweaking: https://github.com/ansible-collections/collection_template/commit/3d991b74c972bf683ac6cd86bdf4980f21c71d6b.

File: .github/workflows/ansible-test.yml
Comment (briantist) [inline .github/workflows/ansible-test.yml:13]: I don't feel strongly about it but should this include Matrix?

**Review by briantist — APPROVED** [review]

File: .github/workflows/ansible-test.yml
Comment (webknjaz) [reply .github/workflows/ansible-test.yml:13]: Dunno, maybe. But in a separate PR. I don't like mixing unrelated changes.

Comment (felixfontein) [reply .github/workflows/ansible-test.yml:13]: I agree to both of you ;)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@webknjaz thanks for this improvement!
@briantist thanks for reviewing!

**Comment by webknjaz** [comment]
Thanks everyone!

**Pull request merged**
